pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -113,6 +115,21 @@ func (authHandler *unionAuthenticationHandler) AuthenticationNeeded(apiClient au
 				w.WriteHeader(http.StatusFound)
 			default:
 				w.WriteHeader(http.StatusUnauthorized)
+				ctx, ok := authHandler.requestContextMapper.Get(req)
+				if !ok {
+					return false, fmt.Errorf("no context found for request to audit")
```

Comment:
Why does this error?

Reply:
It's very serious error if a request does not have associated context, thus we'll error here.